airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/ResetActivity.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.temporal.scheduling.activities;
+
+import io.airbyte.config.StreamDescriptor;
+import io.temporal.activity.ActivityInterface;
+import io.temporal.activity.ActivityMethod;
+import java.util.List;
+import java.util.UUID;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@ActivityInterface
+public interface ResetActivity {
+
+  @Data
+  @NoArgsConstructor
+  @AllArgsConstructor
+  class ResetInput {
+
+    private UUID connectionId;
+
+  }
+
+  @Data
+  @NoArgsConstructor
+  @AllArgsConstructor
+  class GetResetsOutput {
+
+    private List<StreamDescriptor> streamDescriptorList;
+
+  }
+
+  /**
+   * Return the streams being reset for a given connection id
+   */
+  @ActivityMethod
+  GetResetsOutput getStreamResets(UUID connectionId);
+
+  /**
+   * Creates a stream_reset record for each stream descriptor passed in
+   */
+  @ActivityMethod
+  void createStreamResets(UUID connectionId, List<StreamDescriptor> streamDescriptorList);
```

Comment:
How are we imagining that this activity method is being called with this input? I think in the original design, the `streamsToReset` was going to be stored in-memory in the workflow which is easy to do from the `reset` signal method (you just modify the class variable in the signal method). 

But, we changed the design to store these in the db instead. And as far as I know, activities cannot really be directly triggered from signal methods -- signal methods are supposed to just be modifying values in the temporal memory. I think the design was not fully updated to account for this discrepancy.

So I think to make this work, we will need to do one of the following:
- Store the streamsToReset in temporal memory and then execute this activity method somewhere in the workflow where we transfer the in-memory streamsToReset to the db. I think this may be prone to some issues such as if the workflow is restarted for some reason before this activity method is ran, then those in-memory streamsToReset could be lost.
- In the `reset` API endpoint, write the "streams to reset" to the StreamReset table directly (i.e. not through a temporal activity). This way, we are guaranteed that if the `reset` API request is successful, then those streams have been successfully persisted to the db and should be reset on the next run. The downside here is that we will now have multiple places accessing the StreamReset table (the API server and the temporal workflow)

Any thoughts on this @alovew / @benmoriceau / @jdpgrailsdev?